src/org/zaproxy/zap/extension/browserView/BrowserPanel.java

```diff
@@ -88,4 +159,44 @@ private static String toURL(String str) {
             return null;
         }
     }
+    
+    private int getWebsiteHeight(){
+        String script =   "var body = document.body, html = document.documentElement;"
+                        + "Math.max(body.offsetHeight, html.offsetHeight);";
+        return Integer.parseInt(engine.executeScript(script).toString());
+    }
+    
+    private void adjustPanelHeightToWebsite(){
+        final AtomicReference<Integer> webSiteHight = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        
+        Platform.runLater(new Runnable() {
+             @Override 
+             public void run() {
+                 int height = getWebsiteHeight();
+                 webSiteHight.set(height);
+                 latch.countDown();
+             }
+         });
+        
+        try
+        {
+            latch.await(3, TimeUnit.SECONDS);
```

Comment:
It should check the return value, otherwise attempting to set the height after the timeout will lead to `NullPointerException`.

Is 3 seconds an arbitrary value? Or, based on some tests?